nativelib/src/main/scala/scala/scalanative/unsafe/package.scala

```diff
@@ -4,16 +4,18 @@ import java.nio.charset.{Charset, StandardCharsets}
 import scala.language.experimental.macros
 import scalanative.annotation.alwaysinline
 import scalanative.runtime.{Platform, fromRawPtr, intrinsic, libc}
-import scalanative.runtime.Intrinsics.{castIntToRawPtr, castLongToRawPtr}
+import scalanative.runtime.Intrinsics.{
+  castIntToRawPtr,
+  castIntToRawSize,
+  castLongToRawPtr,
+  castLongToRawSize,
+  sizeOfPtr
+}
 import scalanative.unsigned._
 
 package object unsafe {
-
-  /** Int on 32-bit architectures and Long on 64-bit ones. */
-  type Word = Long
-
-  /** UInt on 32-bit architectures and ULong on 64-bit ones. */
-  type UWord = ULong
+  val ptrSize = new USize(sizeOfPtr)
+  val is32    = ptrSize.toInt == 4
```

Comment:
This one can also be moved to `scalanative.meta.LinktimeInfo`. I think it makes a better place for such information